### ISSUE TITLE
[Feature Request] Add encryption codec to SDK

### ISSUE THREAD

**Issue by lorensr** [temporalio/features#339]
### Is your feature request related to a problem? Please describe.

The amount of setup for encryption. Feedback from user:

> I’m working on setting up encryption, following the [example in the TS sdk](https://github.com/temporalio/samples-typescript/blob/main/encryption/README.md). It feels… like a lot of setup. Am I missing something? Is there nothing more plug-and-play that we can use?

### Describe the solution you'd like

> I suppose I would have expected encryption to be something that could be handled by the SDK (perhaps disabled by default), on the client-side. Our workers/clients could provide a key and perhaps an algorithm, and that would be all. A codec server could be deployed on our side, (maybe distributed as a docker image) by Temporal.

**Comment by cretz** [comment]
I think there are too many choices in how to do encryption for this to be in SDK. I think this should be subject to discussion before adding as a "feature". You can't just say "key and algorithm" and go. There is much more to it than that.

**Comment by Quinn-With-Two-Ns** [comment]
I agree with @cretz that it isn't as simple as saying "key and algorithm". That being said there are examples of other [SDKs](https://docs.aws.amazon.com/amazon-s3-encryption-client/latest/developerguide/go-examples.html) for products providing helpers for client side encryption we could take inspiration from.
